streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -893,26 +892,15 @@ private void addStandbyTasks() {
                 newStandbyTasks.put(taskId, partitions);
             }
 
-            if (task != null) {
-                standbyTasks.put(taskId, task);
-                for (TopicPartition partition : partitions) {
-                    standbyTasksByPartition.put(partition, task);
-                }
-                // collect checked pointed offsets to position the restore consumer
-                // this include all partitions from which we restore states
-                for (TopicPartition partition : task.checkpointedOffsets().keySet()) {
-                    standbyTasksByPartition.put(partition, task);
-                }
-                checkpointedOffsets.putAll(task.checkpointedOffsets());
-            }
+            updateStandByTaskMaps(checkpointedOffsets, taskId, partitions, task);
         }
 
         // destroy any remaining suspended tasks
         removeSuspendedStandbyTasks();
 
         // create all newly assigned standby tasks (guard against race condition with other thread via backoff and retry)
         // -> other thread will call removeSuspendedStandbyTasks(); eventually
-        standbyTaskCreator.retryWithBackoff(newStandbyTasks);
+        new StandbyTaskCreator(checkpointedOffsets).retryWithBackoff(newStandbyTasks);
```

Comment:
It is not like this code gets called all the time, so creating a new object is really not an issue. The  interface for `retryWithBackoff` doesn't have or need the checkpointed offsets map and why should it? It doesn't make sense in the other case it is used.
The `checkpointedOffsets` is a new `Map` each time through the rebalance and i think in this case it is nicer to create the `StandbyTaskCreator` as a single shot and not have any state left around between different invocations.